server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -928,32 +927,15 @@ record PluginServiceInstances(
             fileSettingsService,
             threadPool
         );
-        final DiskThresholdMonitor diskThresholdMonitor = new DiskThresholdMonitor(
-            settings,
-            clusterService::state,
-            clusterService.getClusterSettings(),
-            client,
-            threadPool::relativeTimeInMillis,
-            rerouteService
-        );
-        clusterInfoService.addListener(diskThresholdMonitor::onNewInfo);
 
-        final DiscoveryModule discoveryModule = new DiscoveryModule(
+        DiscoveryModule discoveryModule = createDiscoveryModule(
```

Comment:
I'm wondering a bit about naming / semantics of these helper functions. This is mostly minor, but it made it harder for me to follow the code . E.g. looking at registration of modules:
- `createDiscoveryModule`: creates the DiscoveryModule and also registers (adds) it
- `loadPersistentTasksService` / `loadPluginShutdownService`: creates an anonymous module, but doesn't register it yet

Breaking things apart already improves readability by a big deal. But considering the complexity of node construction, it might be nice to follow rather stringent, repetitive patterns.

Reply:
I've tended to use `createXXX` for methods that return the XXX for use by other components, and `loadXXX` for methods that just instantiate things for injection. There's no particular reason why the `load` methods can't add to `modules` themselves, that's just how it fell out

Reply:
Thanks for the clarification 👍  adding (loading) all modules in place might avoid subtile mistakes, but just a nit that can be revisited once we have a plan for the bigger picture here